### ISSUE TITLE
lyd_new_path creates multiple list instances unintentionally

### ISSUE THREAD

**Issue by lukasmacko** [CESNET/libyang#216]
```C
#include <libyang/libyang.h>
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv)
{
   struct ly_ctx *ctx = ly_ctx_new(".");
   ly_ctx_load_module(ctx, "sysrepo-persistent-data", NULL);
   struct lyd_node *node = lyd_new_path(NULL, ctx, "/sysrepo-persistent-data:module[name='example-module']/subscriptions/subscription[type='module-change'][destination-address='addr1'][destination-id='123']", NULL, 0, 0);

   lyd_new_path(node, ctx, "/sysrepo-persistent-data:module[name='example-module']/subscriptions/subscription[type='module-change'][destination-address='addr1'][destination-id='123']/enable-running", NULL, 0, 0);

   lyd_new_path(node, ctx, "/sysrepo-persistent-data:module[name='example-module']/subscriptions/subscription[type='module-change'][destination-address='addr1'][destination-id='123']/event", "apply", 0, 0);
   lyd_new_path(node, ctx, "/sysrepo-persistent-data:module[name='example-module']/subscriptions/subscription[type='module-change'][destination-address='addr1'][destination-id='123']/priority", "0", 0, 0);

   lyd_validate(&node, LYD_OPT_CONFIG, NULL);
   lyd_free_withsiblings(node);
   ly_ctx_destroy(ctx, NULL);
   return 0;
}
```
https://github.com/sysrepo/sysrepo/blob/master/yang/sysrepo-persistent-data.yang

**Comment by lukasmacko** [comment]
The error was introduced https://github.com/CESNET/libyang/commit/7a2a562b891f737768810f50d6d9d7b25100b2eb .

**Comment by michalvasko** [comment]
Hi,
this particular problem should be fixed by 9ba34de6da545d398ca069cb9525fed8d1c7f9df, but it is possible there are still other connected to this one (sysrepo tests still did not pass on my machine, but I failed to find the cause).

Regards,
Michal

**Comment by lukasmacko** [comment]
Hi Michal,

I've fixed the issues on our side currently just in my fork https://github.com/lukasmacko/sysrepo. Almost all of our test passes the remaining issue is https://github.com/CESNET/libyang/issues/218

Regards,
Lukas
